src/SoftekBarcodeAPI.php

```diff
@@ -0,0 +1,37 @@
+<?php
+declare(strict_types=1);
+
+namespace DexproSolutionsGmbh\SoftekBarcodeWrapper;
+
+use DexproSolutionsGmbh\SoftekBarcodeWrapper\Model\SoftekBarcodeConfig;
+use FFI;
+
+class SoftekBarcodeAPI
+{
+    /**
+     * The foreign C API
+     *
+     * @var FFI
+     */
+    protected FFI $interface;
+
+    /**
+     * The persistent instance of the barcode tool kit
+     *
+     * @var FFI\CData
+     */
+    protected FFI\CData $instance;
+
+    public function __construct(SoftekBarcodeConfig $configuration)
+    {
+        // init the c library with his declarations and source code
+        $this->interface = FFI::cdef(file_get_contents($configuration->getHeaderFilePath()), $configuration->getSourceFilePath());
+
+        // create a persistent instance of the interface so that we can work with the same one during operations
+        $this->instance = $this->interface->mtCreateBarcodeInstance();
+
+        // license the instance for further operations
+        $this->interface->mtSetLicenseKey($this->instance, $configuration->getLicenseKey());
```

Comment:
Can any of these calls on FFI function calls fail with an Exception or some return value?

If so, I think our library should explicitly catch those exceptions and re-throw them. Package users should be able to notice (via linter, IDE or code documentation) that calling a method on `SoftekBarcodeAPI` could cause an exception.

Therefore, we should explicitly document

```php
/**
 * @throws SoftekException If the license is invalid or the shared library was not found.
 */
public function __construct(SoftekBarcodeConfig $configuration)
```

Reply:
thanks for your feedback, custom error exceptions was in a locally feature branch i would merge both branches and would update this pull request.

Reply:
Nice.